forest/cli/src/cli/wallet_cmd.rs

```diff
@@ -189,37 +170,18 @@ impl WalletCommands {
                     let addr = address.0.to_string();
                     let default_address_mark = if addr == default { "X" } else { "" };
 
-                    let balance_string =
-                        match wallet_balance((addr.clone(),), &config.client.rpc_token).await {
-                            Ok(balance) => balance,
-                            Err(err) => {
-                                println!("Failed loading the wallet balance");
-                                return Err(handle_rpc_err(err));
-                            }
-                        };
+                    let balance_string = wallet_balance((addr.clone(),), &config.client.rpc_token)
+                        .await
+                        .map_err(handle_rpc_err)?;
 
-                    let balance_int = match balance_string.parse::<BigInt>() {
-                        Ok(balance) => TokenAmount::from_atto(balance),
-                        Err(err) => {
-                            println!(
-                                "Couldn't convert balance {balance_string} to TokenAmount: {err}"
-                            );
-                            continue;
-                        }
-                    };
+                    let balance_int = TokenAmount::from_atto(balance_string.parse::<BigInt>()?);
```

Comment:
It's an internal RPC bug if the balance cannot be parsed. Ideally we would change the RPC type to `TokenAmount` rather than `String` but that's difficult for silly reasons.